### ISSUE TITLE
Invalid f-string conversion type code parses/passes

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ruff#18671]
### Summary

It feels like this code should be rejected, or there should be some sort of lint for this. All three of these f-strings are invalid
```py
f"{1!
r}"
f"{1! r}"
f"{1!\
r}"
```
And raise `SyntaxError: f-string: conversion type must come right after the exclamanation mark`, but it parses/formats with no complaints, even with all lints enabled [playground link](https://play.ruff.rs/f054e9f1-21f8-4b1e-91ad-463f11e54a7f)
(Also I already make a PR in CPython for the typo in the error)

### Version

playground

**Comment by MeGaGiGaGon** [comment]
You can also use this fact to cause code changing formatting/comment deletion [playground](https://play.ruff.rs/8c0ad55f-181c-4379-9491-a50bb34ea8b7)
```py
f"{1! #test
r:}"
f"{1=! #test
r:}"
```
Formats to
```py
f"{
    1!r:  # test
}"
f"{1=!r:}"
```

**Comment by ntBre** [comment]
Thanks! I'll add these to https://github.com/astral-sh/ruff/issues/17412

**Comment by ntBre** [comment]
Oh, these are actually reported by the parser, not the compiler. We can just track them here then.
